content/academy/puppeteer_playwright/proxies.md

```diff
@@ -0,0 +1,180 @@
+
+In this lesson, we'll be learning how to use proxies with Playwright and Puppeteer. This will be demonstrated with a Vietnamese proxy that we got by running [this](https://apify.com/mstephen190/proxy-scraper) proxy-scraping actor on the Apify platform.
+
+## [](#adding-a-proxy) Adding a proxy
+
+First, let's go ahead and create a variable called `proxy` which will point to our proxy server:
+
+> Note that this proxy may no longer be working at the time of reading. If you don't have a proxy to use during this lesson, we recommend using Proxy Scraper for a list of free ones, or checking out [Apify proxy](https://apify.com/proxy)
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+// our proxy server
+const proxy = '103.214.9.13:3128';
+
+const browser = await chromium.launch({ headless: false });
+const page = await browser.newPage();
+await page.goto('https://google.com');
+
+await page.waitForTimeout(10000);
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+// our proxy server
+const proxy = '103.214.9.13:3128';
+
+const browser = await puppeteer.launch({ headless: false });
+const page = await browser.newPage();
+await page.goto('https://google.com');
+
+await page.waitForTimeout(10000);
+await browser.close();
+</marked-tab>
+```
+
+For both Puppeteer and Playwright, the proxy's server URL should be passed into the options of the `launch()` function; however, it's done a bit differently depending on which library you're using.
+
+In Puppeteer, the server must be passed within the **--proxy-server** [Chromium command line argument](https://peter.sh/experiments/chromium-command-line-switches/), while in Playwright, it can be passed into the **proxy** option.
+
+```marked-tabs
+<marked-tab header="Playwright" lang="javascript">
+import { chromium } from 'playwright';
+
+const proxy = '103.214.9.13:3128';
+
+const browser = await chromium.launch({
+    headless: false,
+    // Using the "proxy" option
+    proxy: {
+        // Pass in the server URL
+        server: proxy,
+        
+    },
+});
+const page = await browser.newPage();
+await page.goto('https://google.com');
+
+await page.waitForTimeout(10000);
+await browser.close();
+</marked-tab>
+<marked-tab header="Puppeteer" lang="javascript">
+import puppeteer from 'puppeteer';
+
+const proxy = '103.214.9.13:3128';
+
+// Using the "args" option, which is an array of Chromium command
+// line switches, we pass the server URL in with "--proxy-server"
+const browser = await puppeteer.launch({ headless: false, args: [`--proxy-server=${proxy}`] });
```

Comment:
Multi-lining it, because the important param was partially cut off and one needed to scroll to se it.

```suggestion
const browser = await puppeteer.launch({
    headless: false,
    args: [`--proxy-server=${proxy}`]
});
```